File: snippets/fsharp/System/Version/TryParse/tryparse1.fs
// <Snippet1>
open System

let parseVersion (input: string) =
    match Version.TryParse input with
    | true, ver ->
        printfn $"Converted '{input} to {ver}."
    | _ ->
        printfn $"Unable to determine the version from '{input}'."

[<EntryPoint>]
let main _ =
    let input = "4.0"
    parseVersion input

    let input = "4.0."
    parseVersion input

    let input = "1.1.2"
    parseVersion input

    let input = "1.1.2.01702"
    parseVersion input

    let input = "1.1.2.0702.119"
    parseVersion input

    let input =  "1.3.5.2150000000"
    parseVersion input
    0
// The example displays the following output:
//       Converted '4.0 to 4.0.
//       Unable to determine the version from '4.0.'.
//       Converted '1.1.2 to 1.1.2.
//       Converted '1.1.2.01702 to 1.1.2.1702.
//       Unable to determine the version from '1.1.2.0702.119'.
//       Unable to determine the version from '1.3.5.2150000000'.
// </Snippet1>

File: snippets/csharp/System/Enum/TryParseTEnum/tryparse1.cs
// <Snippet1>
using System;

[Flags] enum CaseSensitiveColors { None = 0, Red = 1, Green = 2, Blue = 4 };

public class CaseSensitiveTryParseExample
{
    public static void Run()
    {
        string[] colorStrings = { "0", "2", "8", "blue", "Blue", "Yellow", "Red, Green" };
        foreach (string colorString in colorStrings)
        {
            CaseSensitiveColors colorValue;
            if (Enum.TryParse(colorString, out colorValue))
                if (Enum.IsDefined(typeof(CaseSensitiveColors), colorValue) | colorValue.ToString().Contains(","))
                    Console.WriteLine("Converted '{0}' to {1}.", colorString, colorValue.ToString());
                else
                    Console.WriteLine($"{colorString} is not an underlying value of the Colors enumeration.");
            else
                Console.WriteLine($"{colorString} is not a member of the Colors enumeration.");
        }
    }
}
// The example displays the following output:
//       Converted '0' to None.
//       Converted '2' to Green.
//       8 is not an underlying value of the Colors enumeration.
//       blue is not a member of the Colors enumeration.
//       Converted 'Blue' to Blue.
//       Yellow is not a member of the Colors enumeration.
//       Converted 'Red, Green' to Red, Green.
// </Snippet1>
